lib/fresh_connection/connection_factory.rb

```diff
@@ -22,14 +23,40 @@ def spec
       @spec ||= build_spec
     end
 
+    # when building a spec from envars, there may be no database.yml file, so
+    # be careful to avoid implicit dependency on it or derived contents.
+
     def build_spec
+      envar_name = 'DATABASE_' + (@group || 'replica').to_s.upcase + '_URL'
+      if (url = ENV[envar_name])      # DATABASE_REPLICA{,1,2}_URL
+        spec = build_spec_from_url(url)
+        set_group_spec_and_url(spec, url) if @group
+        spec
+      else
+        build_spec_from_config
+      end
+    end
+
+    def set_group_spec_and_url(spec, url)
+      if @group
+        configs = (ar_spec.config || {}).symbolize_keys
+        configs[@group] ||= {}
+        configs[@group]["url"] = url
```

Comment:
When ActiveRecord opens a connection, if there is a `"url"` key, it will resolve the components from parsing the associated value. 

The `configs[@group]` is a hash, which is passed by reference, so assigning "url" key makes it effective for a later AR `establish_connection`